### PR TITLE
[NTUSER] Indirect fix to SetProcessDefaultLayout

### PR DESCRIPTION
**Pull request by peterooch**
## Purpose

~~Added a check in User32CreateWindowEx for parentless windows,
that checks the default layout direction, and if its LAYOUT_RTL add the WS_EX_LAYOUTRTL flag to the extended window styles.~~

Reenable a commented out check that made SetProcessDefaultLayout not to affect UI layout.

Now all the reactos applications that have SetProcessDefaultLayout calls to mirror the layout get mirrored.
(albeit to mixed results)

[How wine checks for default process layout](https://github.com/wine-mirror/wine/blob/da5112c74313b5236183135204cab38a8b0169ed/dlls/user32/win.c#L1437)

### CODE REVIEW

File: win32ss/user/user32/windows/window.c [hunk 1]
```diff
@@ -283,6 +283,17 @@ User32CreateWindowEx(DWORD dwExStyle,
         plstrClassVersion = &lstrClassVersion;
     }
 
+    /* Check process default layout */
+    if (!hWndParent)
```
Comment (HBelusca) [inline win32ss/user/user32/windows/window.c]: As done on Windows, setting the LAYOUTRTL flag should be done in the kernel.
After analysis, a good place would be in co_UserCreateWindowEx() or, better, in IntCreateWindow().

**Comment by HBelusca** [comment]
wow! ^^

**Comment by HBelusca** [comment]
Note:
https://svn.reactos.org/svn/reactos?view=revision&revision=62613
https://svn.reactos.org/svn/reactos?view=revision&revision=62679
Testman run with the patch:
https://www.reactos.org/testman/compare.php?ids=61131,61136,61140
see middle column (Revision df383a3) : user32:win test gets more failures with the patch enabled.

**Comment by HBelusca** [comment]
Either the user32:win test is fundamentally wrong, or there is other bugs in win32k, but we still get additional errors in user32:win test with the latest commit:
https://reactos.org/testman/compare.php?ids=61308,61312,61316,61319

**Comment by HBelusca** [comment]
With this new commit, we still have errors but less than without it:
https://reactos.org/testman/compare.php?ids=61316,61319,61324,61325,61326
First, third and fourth columns are tests without the PR patches applied (user32:win : 269 errors);
Second column are tests with the PR patches **BUT WITHOUT** the last patch (user32:win : 471 errors);
Fifth column are tests with the PR patches **INCLUDING** the last patch (user32:win : 444 errors).

**Comment by peterooch** [comment]
I'll try to figure how wine does handle this behavior and i'll open a new PR, sorry about this git-mess

**Pull request closed without merging**